### PR TITLE
Update xlang kinesis to v2

### PR DESCRIPTION
**Pull request by damccorm**
We are actively removing the v1 AWS IOs, so we need to move the x-lang version to use v2. I tried to make this change in as non-breaking a way as possible, but there are differences across the IOs (e.g. not supporting producer properties), which should cause the current users to break/respond.

Part of #33430

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://github.com/apache/beam/blob/master/CONTRIBUTING.md#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

File: sdks/python/apache_beam/io/external/xlang_kinesisio_it_test.py [hunk 1]
```diff
@@ -64,7 +64,7 @@
   DockerContainer = None
 # pylint: enable=wrong-import-order, wrong-import-position, ungrouped-imports
 
-LOCALSTACK_VERSION = '0.11.3'
+LOCALSTACK_VERSION = '3.8.1'
```
Comment (damccorm) [inline sdks/python/apache_beam/io/external/xlang_kinesisio_it_test.py:67]: New aws ios didn't seem to work with old localstack, so I had to update this and some of the other things in this file.

**Comment by damccorm** [comment]
@aromanenko-dev FYI, here is the first PR to remove V1 IOs. This is the most involved, the others are mostly just cutting out code.

**Comment by damccorm** [comment]
assign set of reviewers

**Review by Abacn — APPROVED** [review]
Thank you!

is there a plan to remove https://github.com/apache/beam/blob/master/sdks/java/io/kinesis/expansion-service/ ?

**Comment by damccorm** [comment]
> Thank you!
> 
> is there a plan to remove https://github.com/apache/beam/blob/master/sdks/java/io/kinesis/expansion-service/ ?

Yes, https://github.com/apache/beam/pull/33456 . I split this PR and another one out since they have more significant changes and that one can be a deletion-only PR

**Pull request merged**